internal/policy/policy_test.go

```diff
@@ -337,3 +336,95 @@ func TestGetStateForCommit(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, firstState, state)
 }
+
+func TestFindDelegationEntry(t *testing.T) {
+	t.Run("no delegation", func(t *testing.T) {
+		state := createTestStateWithPolicy(t)
+
+		entry, err := state.findDelegationEntry("random")
+		assert.ErrorIs(t, err, ErrDelegationNotFound)
+		assert.Equal(t, entry, tuf.Delegation{})
+	})
+	t.Run("simple delegation", func(t *testing.T) {
+		state := createTestStateWithPolicy(t)
+		state.DelegationEnvelopes = map[string]*sslibdsse.Envelope{}
+
+		topLevelDelegations := &tuf.Delegations{Roles: []tuf.Delegation{{Name: "1"}}}
+		topLevelMetadata := tuf.TargetsMetadata{Delegations: topLevelDelegations}
+
+		topLevelEnv, err := dsse.CreateEnvelope(topLevelMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		state.TargetsEnvelope = topLevelEnv
+		state.DelegationEnvelopes["targets"] = topLevelEnv
+
+		secondLevelDelegations := &tuf.Delegations{}
+		secondLevelMetadata := tuf.TargetsMetadata{Delegations: secondLevelDelegations}
+		secondLevelEnv, err := dsse.CreateEnvelope(secondLevelMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		state.DelegationEnvelopes["1"] = secondLevelEnv
+
+		entry, err := state.findDelegationEntry("1")
+		assert.Nil(t, err)
+		assert.Equal(t, tuf.Delegation{Name: "1"}, entry)
+	})
+	t.Run("delegation with multiple roles", func(t *testing.T) {
+		state := createTestStateWithPolicy(t)
```

Comment:
same general comment here, I don't love modifying `state` like this as `findDelegationEntry` assumes `state.Verify()` has passed in the invoking function. It's not the end of the world but a clean helper with real policies would be nice. It could be as simple as targets delegating to role A and B, with role A delegating to role C and D for some simple branch / tag / file rules.

Reply:
I have moved it to a different helper func in helpers_test.go